implementations/micrometer-registry-influx/src/main/java/io/micrometer/influx/InfluxMeterRegistry.java

```diff
@@ -189,14 +189,19 @@ protected void publish() {
         return Stream.empty();
     }
 
-    private Stream<String> writeFunctionTimer(FunctionTimer timer) {
-        Stream<Field> fields = Stream.of(
-                new Field("sum", timer.totalTime(getBaseTimeUnit())),
-                new Field("count", timer.count()),
-                new Field("mean", timer.mean(getBaseTimeUnit()))
-        );
-
-        return Stream.of(influxLineProtocol(timer.getId(), "histogram", fields));
+    // VisibleForTesting
+    Stream<String> writeFunctionTimer(FunctionTimer timer) {
+        double sum = timer.totalTime(getBaseTimeUnit());
+        double mean = timer.mean(getBaseTimeUnit());
+        if (Double.isFinite(sum) && Double.isFinite(mean)) {
```

Comment:
In the examples given in the issue, the `sum` had a finite value but the `mean` didn't. I'm wonder if it would make sense to exclude only the `mean` field in that case. I'm not sure what implications this might have in InfluxDB for the `mean` field to be there sometimes but not others.

Reply:
@shakuzen Thanks for the feedback!

I think that `mean` should have used the same snapshot as the `sum` for consistency but uses its own snapshot due to the current implementation. I just chose to drop them all as a whole, but if retaining part of it is better, I could split the two. I don't know the implications of doing so, either as I'm not an InfluxDB user.

Reply:
I think keeping the sum and count even if the mean isn't available is better. I did some ad hoc testing locally and I don't think it is a problem for queries - no data will be returned if querying for the `mean` field during that time, but other fields can be queried during that time.

Reply:
@shakuzen Does it make sense to do the same to `ElasticMeterRegistry` for consistency?

See https://github.com/micrometer-metrics/micrometer/pull/2228

Reply:
It may, but I'd like to setup a test with that to ensure it doesn't break anything for Elasticsearch users on querying, so we can probably follow-up on that. I want to get the next set of patch releases out ASAP.

Reply:
@shakuzen Thanks for the quick feedback! I updated as suggested.

Reply:
Thank you for the quick turnaround. Always appreciated!